data/auth-router.js

```diff
@@ -0,0 +1,41 @@
+const bcrypt = require('bcryptjs');
+const Users = require('./users.js');
+const router = require('express').Router();
+
+router.post('/register', (req, res) => {
+    let user = req.body;
+  
+  user.password= bcrypt.hashSync(user.password, 10); 
+  
+    Users.add(user)
+      .then(saved => {
+        req.session.user = saved;
+        res.status(201).json(saved);
+      })
+      .catch(error => {
+        res.status(500).json(error);
+      });
+  });
+
+  router.post('/api/login', (req, res) => {
+    let { username, password } = req.body;
+  
+    password= bcrypt.hashSync(user.password, 10); 
```

Comment:
You probably wanna use the plain password without hashing it, as bcrypt.compareSync() func will hash the password and compare to the second argument you passed in.